File: JavaScript Fundamentals/ArraysHomework/Problem 5. Selection sort/SelectionSort.js
var arr = [3, 89, -45, 23, 98, 21345, -12, 0, -443, -87, -90, 29, 165],
    leftElement,
    rightElement,
    i,
    len,
    temp;

for (leftElement = 0, len = arr.length; leftElement < len; leftElement += 1) {
    for (var rightElement = leftElement + 1; rightElement < len; rightElement += 1) {
        if (arr[leftElement] > arr[rightElement]) {
            temp = arr[rightElement];
            arr[rightElement] = arr[leftElement];
            arr[leftElement] = temp;
        }
    }
}

console.log(arr.join(', '));

File: CSharp Part 2/ArraysHomework/Problem 07. Selection sort/SelectionSort.cs
// Problem 7. Selection sort
// Sorting an array means to arrange its elements in increasing order. Write a program to sort an array.
// Use the Selection sort algorithm: Find the smallest element, move it at the first position, 
// find the smallest from the rest, move it at the second position, etc.

using System;
using System.Linq;

class SelectionSort
{
    static void Main()
    {
        Console.WriteLine("This program sorts the elements of an array using the selection sort method.");
        Console.Write("Please enter the length of the array: ");
        int N = int.Parse(Console.ReadLine());
        int[] array = new int[N];
        for (int i = 0; i < N; i++)
        {
            Console.Write("element[{0}] = ", i);
            array[i] = int.Parse(Console.ReadLine());
        }

        int temp = 0;
        int[] sortedArray = new int[array.Length];

        for (int i = 0; i < array.Length; i++)
        {
            temp = Array.IndexOf(array, array.Min());
            sortedArray[i] = array[temp];
            array[temp] = int.MaxValue;

        }
        Console.WriteLine("Result: " + string.Join(", ", sortedArray));
    }
}